bot/exts/recruitment/talentpool/_cog.py

```diff
@@ -338,6 +348,97 @@ async def nominate_command(self, ctx: Context, user: MemberOrUser, *, reason: st
+                ephemeral=True
+            )
+            return
+
+        if isinstance(message.author, Member) and any(role.id in STAFF_ROLES for role in message.author.roles):
+            await interaction.response.send_message(
+                ":x: Nominating staff members, eh? Here's a cookie :cookie:",
+                ephemeral=True
+            )
+            return
+
+        # Search for an existing nomination, if we find one we will append the message onto the existing reason.
+        maybe_nom = await self.api.get_nomination_reason(message.author.id, interaction.user.id)
+
+        if maybe_nom:
+            nom, reason = maybe_nom
+
+            new_reason = f"""
+            {reason}
+
+            {message.jump_url}
+            """
+
+            await self.api.edit_nomination_entry(nom.id, actor_id=interaction.user.id, reason=new_reason)
+
+            await interaction.response.send_message(
+                ":white_check_mark: Appended message to your existing nomination",
+                ephemeral=True
+            )
+        else:
+            # If no nomination is found, we will create a new one with the selected message
+            reason = f"Nominated from: {message.jump_url}"
+            try:
+                await self.api.post_nomination(message.author.id, interaction.user.id, reason)
+
+                await interaction.response.send_message(
+                    f":white_check_mark: The nomination for {message.author.mention} has been added to the talent pool",
+                    ephemeral=True
+                )
+            except ResponseCodeError as e:
+                match (e.status, e.response_json):
+                    case (400, {"user": _}):
+                        await interaction.response.send_message(
+                            f":x: {message.author.mention} can't be found in the database tables.",
+                            ephemeral=True
+                        )
+                        return
+
+                raise e
+
+    async def _nominate_context_error(
+            self,
+            interaction: discord.Interaction,
+            error: app_commands.AppCommandError
+    ) -> None:
+        """Handle any errors that occur with the nomination context command."""
+        if isinstance(error, app_commands.errors.MissingAnyRole):
+            await interaction.response.send_message(
+                ":x: You do not have permission to use this command", ephemeral=True
+            )
+        else:
+            message = (
+                f":x: An unexpected error occured, Please let us know!\n\n"
+                f"```{error.__class__.__name__}: {error}```"
+            )
+
+            if not interaction.response.is_done():
+                await interaction.response.send_message(message, ephemeral=True)
+            else:
+                await interaction.followup.send(message, ephemeral=True)
+
+            with push_scope() as scope:
```

Comment:
This is deprecated as of sentry_sdk 2.0. If you want to change this in this PR then:

```suggestion
            with sentry_sdk.new_scope() as scope:
```

Reply:
I'll change it here and add a commit to change it in the other places